Improve driver deprecation UX

jeff303 opened the pull request:
Make the DriverWarning component select the older driver when a "link" is clicked from the new one

And do the reverse (going from deprecated to new)

Making styling match mockup

jeff303 committed:
Improve driver deprecation UX

Reworking the DriverWarning component to be able to change the engine itself by firing the event (so that the "go to older/newer version" links actually work)

Styling fixes to match Figma mockup

kulyk reviewed (changes requested):
Great work @jeff303 👏  I'm sorry, the frontend's styling side is confusing right now: there are CSS files, CSS modules, utility classes, `Flex` / `Box`, and styled-components. We're trying not to use anything except `styled-components` for new code and refactor existing things step-by-step. Could you please refactor the visual components? I've inlined a few things with refactoring examples, I hope they can help

jeff303 committed:
PR feedback from Anton

Remove all grid-styled usage

Remove all styling (except class names) from DriverWarning jsx file, move to the styled-component

jeff303 commented on frontend/src/metabase/css/core/colors.css:
```diff
@@ -196,6 +196,11 @@
+.text-dark-orange-hover:hover {
+  color: var(--color-accent5);
+}
+
```
Should be fixed in the latest commit.

jeff303 commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -73,16 +90,21 @@ function DriverWarning({ engine, ...props }) {
-        <Warnings
-          className="mx2 text-gold"
-          warnings={[tooltipWarning]}
+      <IconContainer
+        align="center"
+        justify="center"
+        className="flex-no-shrink circular"
+        hasCircle={hasCircle}
+      >
```
Removed all `grid-styled` usage

kulyk replied:
Thank you! You can now remove `align` and `justify` props. Also we can remove CSS classes and paste the CSS needed into styled component definition

jeff303 commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -8,50 +8,67 @@ import {
-        {t`We recommend that you upgrade to the new ${
-          allEngines[engine]["driver-name"]
-        } driver, which is faster and more reliable.`}
+      <p className="text-medium m0" style={{ marginTop: 8, marginBottom: 8 }}>
```
Tried adding this in the latest commit, but I must not have done it correctly since there is no margin added between paragraphs (see latest screenshots)

jeff303 commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -73,16 +90,21 @@ function DriverWarning({ engine, ...props }) {
+        <Icon
           size={20}
+          name={supersedes ? "info" : "warning"}
+          className={supersededBy && "text-dark-orange"}
```
I reworked things a bit here so this is no longer used.

jeff303 commented:
@kulyk , thanks for the review.  I added another commit to hopefully address most of what you wrote.  However, the styling still isn't quite correct.  Updated screenshots follow.  Notice there is no margin between the paragraphs despite trying to add it.

<img width="1084" alt="supersedes_view" src="https://user-images.githubusercontent.com/3521562/129979404-51ff1d8a-43ea-4414-ad03-5eba4919cc8b.png">
<img width="1201" alt="superseded_by_view" src="https://user-images.githubusercontent.com/3521562/129979405-0baa2786-b0fa-4c58-b04c-ed650d4f737e.png">

kulyk commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -8,51 +8,70 @@ import {
-      } now.`}</p>
+      </WarningParagraph>
+
+      <WarningParagraph className="text-medium m0" hasMargin={true}>
```
1. You can just use `hasMargin` and it will be the same thing as if you pass `hasMargin={true}` directly
2. The margins are missing because thr `m0` CSS class overwrites it, so you can remove it
3. Please remove the `text-medium` class too, you can replace it with `color: ${color("text-medium")};` in the styled component definition

jeff303 replied:
Fixed all in next commit.

kulyk commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -8,51 +8,70 @@ import {
+        {t`The old driver has been deprecated and will be removed in the next release. If you really
+      need to use it, you can `}
+        &nbsp;
+        <a
+          className="text-brand text-bold"
+          onClick={() => onChangeEngine(supersedesDriver)}
+        >{t`find it here`}</a>
```
Can be a styled component too

jeff303 replied:
Fixed in next commit.

kulyk commented on frontend/src/metabase/components/DriverWarning/DriverWarning.jsx:
```diff
@@ -8,51 +8,70 @@ import {
+      <WarningParagraph className="text-medium m0" hasMargin={true}>
+        {t`We recommend that you upgrade to the`}
+        &nbsp;
+        <a
+          className="text-brand text-bold"
+          onClick={() => onChangeEngine(engine)}
+        >{t`new ${allEngines[engine]["driver-name"]} driver`}</a>
+        {t`, which is faster and more reliable.`}
+      </WarningParagraph>
```
Can be a styled component too

jeff303 replied:
Fixed in next commit.

jeff303 committed:
More PR feedback from Anton

Move more stuff to the styled components

kulyk reviewed (approved):
LGTM! Thank you for addressing every review comment so diligently, Jeff. Great work!

jeff303 committed:
Add backend for warning banner

Implement db-ids-with-deprecated-drivers API endpoint for fetching DB IDs using deprecated drivers

Add some plugin test init code to properly handle the test legacy/new drivers

Adding test for new API

Adding new entry in `MetabaseApi` frontend component for new API method

jeff303 committed:
Add missing FormMessage and FormSubmit back to Form

jeff303 committed:
Fixing Cypress failures by reintroducing the weird `Next` button in new DB scenario
